VirtoCommerce.Storefront/Domain/Lists/WishlistConverter.cs

```diff
@@ -0,0 +1,86 @@
+﻿using VirtoCommerce.Storefront.Model;
+using VirtoCommerce.Storefront.Model.Common;
+using VirtoCommerce.Storefront.Model.Lists;
+using VirtoCommerce.Storefront.Model.Security;
+using cartDto = VirtoCommerce.Storefront.AutoRestClients.CartModuleApi.Models;
+
+namespace VirtoCommerce.Storefront.Domain.Lists
+{
+    public static class WishlistConverterConverterExtension
+    {
+        public static WishlistConverter WishlistConverterInstance
+        {
+            get
+            {
+                return new WishlistConverter();
+            }
+        }
+
+        public static Wishlist ToWishlist(this cartDto.ShoppingCart cartDto, Currency currency, Language language, User user)
+        {
+            return WishlistConverterInstance.ToWishlist(cartDto, currency, language, user);
+        }
+
+        public static cartDto.ShoppingCartSearchCriteria ToSearchCriteriaDto(this WishlistSearchCriteria criteria)
+        {
+            return WishlistConverterInstance.ToSearchCriteriaDto(criteria);
+        }
+    }
+
+    public partial class WishlistConverter : CartConverter
+    {
```

Comment:
I would  suggest use there a hack with Json.Convert to convert base class (Cart) to derived class (Wishlist) instead of this assignments

Reply:
I'd rather not - it's a bit slow, need to add many constructors to cart, cart line item, etc